mlir/lib/Conversion/GPUToLLVMSPV/GPUToLLVMSPV.cpp

```diff
@@ -286,30 +291,94 @@ struct GPUShuffleConversion final : ConvertOpToLLVMPattern<gpu::ShuffleOp> {
            val == getSubgroupSize(op);
   }
 
+  static bool needsBitCastOrExt(gpu::ShuffleOp op) {
+    Type type = op.getType(0);
+    return isa<BFloat16Type>(type) || type.isInteger(1);
+  }
+
+  static Type getBitCastOrExtTy(Type oldTy,
+                                ConversionPatternRewriter &rewriter) {
+    return TypeSwitch<Type, Type>(oldTy)
+        .Case<BFloat16Type>([&](auto) { return rewriter.getIntegerType(16); })
+        .Case<IntegerType>([&](auto intTy) -> Type {
```

Comment:
nit: I don't see why these need to be generic over the argument type
```suggestion
        .Case<BFloat16Type>([&](Type) { return rewriter.getIntegerType(16); })
        .Case<IntegerType>([&](IntegerType intTy) -> Type {
```

Reply:
I am not sure that would compile.

```cpp
        .Case([&](BFloat16Type) { return rewriter.getIntegerType(16); })
        .Case([&](IntegerType intTy) -> Type {
```

Is nicer IMO tho